packages/context/src/binding.ts

```diff
@@ -54,14 +71,19 @@ export class Binding {
 
   private readonly _key: string;
   private _tags: Set<string> = new Set();
+  private _scope: BindingScope;
+  private _cache: BoundValue;
+  private _getValue: (ctx?: Context) => BoundValue | Promise<BoundValue>;
 
   // For bindings bound via toClass, this property contains the constructor
   // function
   public valueConstructor: Constructor<BoundValue>;
 
-  constructor(_key: string, public isLocked: boolean = false) {
+  constructor(_key: string, public isLocked: boolean = false,
+    scope: BindingScope = BindingScope.TRANSIENT) {
     Binding.validateKey(_key);
     this._key = _key;
+    this._scope = scope;
```

Comment:
FWIW, I think this line can be omitted and the property can be declared via a constructor argument:

```ts
private _scope: BindingScope = BindingScope.TRANSIENT;
```

Also I think the constructor has too many parameters at this point, we should stop adding more parameters and provide fluent API for setting these extra flags instead (I see you already have `inScope` method for that).